var/spack/repos/builtin/packages/esys-particle/package.py

```diff
@@ -0,0 +1,59 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class EsysParticle(CMakePackage):
+    """ESyS-Particle is Open Source software for particle-based numerical
+    modelling. The software implements the Discrete Element Method (DEM),
+    a widely used technique for modelling processes involving large
+    deformations, granular flow and/or fragmentation."""
+
+    homepage = "https://launchpad.net/esys-particle"
+    url      = "https://launchpadlibrarian.net/539636757/esys-particle-3.0-alpha.tar.gz"
+
+    maintainers = ['snehring']
+
+    version('3.0-alpha', sha256='4fba856a95c93991cacb904e6a54a7ded93558f7adc8c3e6da99bc347843a434')
+
+    depends_on('mpi')
+    depends_on('boost@1.71.0+python')
+    depends_on('python@3.6:')
+    depends_on('py-setuptools', type='build')
+
+    def patch(self):
+        if self.spec.satisfies('@3.0-alpha'):
```

Comment:
You should be able to use `@when` to conditionally patch:
```suggestion
    @when('@3.0-alpha')
    def patch(self):
```
(and then deindent the files below). I would also recommend replacing your long list below with
```python
files = find('.', 'CMakeLists.txt')
```
where `find` is imported from `llnl.util`.